packages/core/src/amazonqFeatureDev/session/session.ts

```diff
@@ -173,13 +171,13 @@ export class Session {
             const content = await this.config.fs.readFile(uri)
             const decodedContent = new TextDecoder().decode(content)
 
-            await fs.mkdir(path.dirname(absolutePath))
-            await fs.writeFile(absolutePath, decodedContent)
+            await FileSystemCommon.instance.mkdir(path.dirname(absolutePath))
+            await FileSystemCommon.instance.writeFile(absolutePath, decodedContent)
         }
 
         for (const filePath of this.state.deletedFiles?.filter(i => !i.rejected) ?? []) {
             const absolutePath = path.join(filePath.workspaceFolder.uri.fsPath, filePath.relativePath)
-            await fs.delete(absolutePath)
+            await FileSystemCommon.instance.delete(absolutePath)
```

Comment:
? why

Reply:
otherwise the mock of the test doesn't work

Reply:
why are you using mocks instead of creating actual files (which tests real codepaths instead of mocks)?